src/misc/fs_utils.cpp

```diff
@@ -22,16 +22,39 @@
 
 #include <algorithm>
 #include <cassert>
+#include <cerrno>
 #include <chrono>
+#include <cstring>
 #include <fstream>
 
+// #include "cross.h"
 #include "checks.h"
 #include "dos_inc.h"
 #include "dos_system.h"
 #include "std_filesystem.h"
 
 CHECK_NARROWING();
 
+bool check_fseek(const char* module_name, const char* filename, FILE* stream,
+                 const long long offset, const int whence)
+{
+	assert(stream);
+	if (cross_fseeko(stream, offset, whence) == 0) {
+		return true;
+	}
+
+	assert(module_name);
+	assert(filename);
+	LOG_ERR("%s: Failed seeking to byte %lld in file '%s': %s",
+	        module_name,
+	        offset,
+	        filename,
+	        strerror(errno));
+	fclose(stream);
```

Comment:
Isn't this information only useful for devs? Seeks offset, error codes, etc. Would it be better to turn this into an ifdef'd out debug level message, and in the else branch of the ifdef we'd just log an error saying "Error reading file..." or something?

Reply:
`strerror` turns the C error code into something human readable like "permissions error" or "file not found".  As long as this isn't flooding the logs, more info is usually good IMO.  The user doesn't have to know what it means but it helps us when they provide detailed logs on a tricky bug we're trying to fix.

Reply:
Yup; what @weirddan455 said.

These seeks are coming from user-provided files, and they'll be able to see the path and issue, so definitely worth while. 

We tend to try to pass on strerror whenever C functions can fail on user data (so this is just keeping with that trend).

Reply:
Yeah okay, sounds reasonable then. I thought we're logging error codes or something... Less convinced having seek offsets in the logs is a good idea, though, but whatever 😄